outbound/index.js

```diff
@@ -281,6 +281,7 @@ exports.send_trans_email = function (transaction, next) {
                 delivery_queue.push(hmail);
             }
 
+            transaction.qfiles = qfiles;
```

Comment:
Sorry to take so long to get to this review...

Anyway we need to push onto this array every time process_delivery finishes, rather than do it at the end. Otherwise we have to wait for them all to finish that function.